sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -375,6 +375,55 @@ case class GpuSum(child: Expression, resultType: DataType)
     Aggregation.sum().onColumn(inputs.head._2)
 }
 
+case class GpuPivotFirst(
+  pivotColumn: Expression,
+  valueColumn: Expression,
+  pivotColumnValues: Seq[Any],
+  mutableAggBufferOffset: Int = 0,
+  inputAggBufferOffset: Int = 0) extends GpuDeclarativeAggregate {
+
+  val valueDataType = valueColumn.dataType
+
+  override val dataType: DataType = valueDataType
+  override val nullable: Boolean = false
+
+  val pivotColAttr = pivotColumnValues.map(a => {
+    // If `a` is null, then create an AttributeReference for null column.
+    if (a == null) AttributeReference(GpuLiteral(null, valueDataType).toString, valueDataType)()
+    else AttributeReference(a.toString, valueDataType)()
+  })
+
+  override lazy val inputProjection: Seq[Expression] = {
+    val expr = pivotColumnValues.map(a => {
+      if (a == null) {
+        GpuIf(GpuIsNull(pivotColumn), valueColumn, GpuLiteral(null, valueDataType))
+      } else {
+        GpuIf(GpuEqualTo(pivotColumn, GpuLiteral(a, pivotColumn.dataType)),
+          valueColumn, GpuLiteral(null, valueDataType))
+      }
+    })
+    expr
+  }
+
+  override lazy val updateExpressions: Seq[GpuExpression] = {
+    val updateArray = pivotColAttr.map(a => new CudfLastExcludeNulls(a))
```

Comment:
Could you define in this aggregate's class header what the pivot looks like at each stage? I don't think we need to rehash everything that Spark has, but something quick showing why `CudfLastExcludeNulls` works here and in `mergeExpressions`.